src/sgl/core/bc_codec.cpp

```
@@ -0,0 +1,1073 @@
+    case BCComponentType::uint16:
+        return reinterpret_cast<const uint16_t*>(pixel)[channel] / 65535.0f;
+    case BCComponentType::int16:
+        return std::max(-1.0f, reinterpret_cast<const int16_t*>(pixel)[channel] / 32767.0f);
+    case BCComponentType::float16:
+        return math::float16_to_float32(reinterpret_cast<const uint16_t*>(pixel)[channel]);
+    case BCComponentType::uint32:
+        return static_cast<float>(reinterpret_cast<const uint32_t*>(pixel)[channel] / 4294967295.0);
+    case BCComponentType::int32:
+        return std::max(-1.0f, static_cast<float>(reinterpret_cast<const int32_t*>(pixel)[channel] / 2147483647.0));
+    case BCComponentType::float32:
+        return reinterpret_cast<const float*>(pixel)[channel];
```

Cppcheck (2.21.0)

[portability] Casting between const unsigned char * and const float * which have an incompatible binary data representation.

(invalidPointerCast)